site/app/templates/course/CourseMaterials.twig

```diff
@@ -181,10 +181,14 @@
 {% endmacro %}
 
 <script>
+    function setFolderCookie(folderPath, id, open) {
+        const folderData = JSON.parse(Cookies.get('cm_folder_data') || '{}');
+        folderData[folderPath] = { id, open };
+        Cookies.set('cm_folder_data', JSON.stringify(folderData));
+    }
+
     function toggleFoldersOpen(){
-        {% for file in material_list %}
-        Cookies.remove("{{ file.getPath }}".substr(0,'{{ file.getPath }}'.lastIndexOf("/")));
-        {% endfor %}
+        Cookies.remove('cm_folder_data');
         Cookies.set("foldersOpen", toggleCMFolders());
     }
```

Comment:
It would be a good idea to move these functions to a `.js` file so they can be linted.

Reply:
I second this